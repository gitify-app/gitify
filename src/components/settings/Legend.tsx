import type { Icon } from '@primer/octicons-react';
import type { FC } from 'react';

interface ILegend {
  icon: Icon;
  children: string;
}

export const Legend: FC<ILegend> = (props) => {
  return (
    <legend
      id={props.children.toLowerCase().replace(' ', '-')}
      className="mb-1 mt-2 font-semibold flex items-center"
    >
      <props.icon className="mr-2" />
      {props.children}
    </legend>
  );
};
