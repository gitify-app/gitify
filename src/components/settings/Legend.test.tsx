import { PersonFillIcon } from '@primer/octicons-react';
import { render } from '@testing-library/react';
import { Legend } from './Legend';

describe('routes/components/settings/Legend.tsx', () => {
  it('should render the legend', async () => {
    const { container } = render(<Legend icon={PersonFillIcon}>Legend</Legend>);

    expect(container).toMatchSnapshot();
  });
});
