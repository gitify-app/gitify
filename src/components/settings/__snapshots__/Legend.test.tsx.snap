// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`routes/components/settings/Legend.tsx should render the legend 1`] = `
<div>
  <legend
    class="mb-1 mt-2 font-semibold flex items-center"
    id="legend"
  >
    <svg
      aria-hidden="true"
      class="mr-2"
      fill="currentColor"
      focusable="false"
      height="16"
      style="display: inline-block; user-select: none; vertical-align: text-bottom; overflow: visible;"
      viewBox="0 0 16 16"
      width="16"
    >
      <path
        d="M4.243 4.757a3.757 3.757 0 1 1 5.851 3.119 6.006 6.006 0 0 1 3.9 5.339.75.75 0 0 1-.715.784H2.721a.75.75 0 0 1-.714-.784 6.006 6.006 0 0 1 3.9-5.34 3.753 3.753 0 0 1-1.664-3.118Z"
      />
    </svg>
    Legend
  </legend>
</div>
`;
